Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,170 @@ This document includes four main sections:
       PEP written by Marc-André Lemburg.
 
 
+.. We use the following practises for SQL code:
+   - UPPERCASE for keywords
+   - snake_case for schema
+   - single quotes for string literals
+   - singular for table names
+   - if needed, use double quotes for table and column names
+
 .. _sqlite3-tutorial:
 
 Tutorial
 --------
 
-To use the module, start by creating a :class:`Connection` object that
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+In this tutorial, you will learn the basics of the :mod:`!sqlite3` API
+by creating a database of Monty Python movies.
+It assumes a fundamental understanding of database concepts,
+including `cursors`_ and `transactions`_.
+
+First, you'll need to create a new database to hold the movie data
+and open a database connection to allow :mod:`!sqlite3` to work with it.
+Call the :func:`sqlite3.connect` function to
+to create a connection to a database :file:`tutorial.db`
+in the current working directory,
+implicitly creating it if it does not exist::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+The returned :class:`Connection` object ``con``
+represents the connection to the database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+you'll use a database cursor.
+Call :meth:`con.cursor() <Connection.cursor>` to create the :class:`Cursor`::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
-
-   # Insert a row of data
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
+Now you'll create a database table ``movie`` with columns for title,
+release year, and review score.
+For simplicitly, just use column names in the table declaration:
+thanks to the `flexible typing`_ feature of SQLite,
+specifying the data types is optional.
+Execute the ``CREATE TABLE`` statement
+by calling :meth:`con.execute() <Cursor.execute>`::
+
+   cur.execute("CREATE TABLE movie(title, year, score)")
+
+.. Ideally, we'd use sqlite_schema instead of sqlite_master below,
+   but earlier versions of SQLite do not recognise that variant.
+
+You can verify that the new table has been created by querying
+the ``sqlite_master`` table built-in to SQLite.
+It should contain an entry for the ``movie`` table definition.
+Execute that query by calling :meth:`cur.execute(...) <Cursor.execute>`,
+store the result in a variable ``res``,
```

Comment:
Just a nit, but technically we are not "storing the result in a variable".  This works for languages like C where variables correspond to specific memory location (where you store the data), but in Python variables are just references to an object.

In addition, people reading this tutorial should already be familiar with how functions/methods work, and the fact that they return a value that you can assign to a variable, so you can probably omit this.

Reply:
I thought about mentioning this, but at least at the time I wasn't sure if it was worth complicating things for beginners. But it does irk me too.

The reason it is mentioned is because it is referred to in the next clause, so we need to say *something*. What about 

```suggestion
assign the result to ``res``,
```

That's both simpler and more correct.